third_party/libtock-drivers/src/nfc.rs

```diff
@@ -0,0 +1,144 @@
+use crate::util;
+use core::cell::Cell;
+use libtock_core::{callback, syscalls};
+
+const DRIVER_NUMBER: usize = 0x30003;
+
+mod command_nr {
+    pub const TRANSMIT: usize = 1;
+    pub const RECEIVE: usize = 2;
+    pub const EMULATE: usize = 3;
+    pub const CONFIGURE: usize = 4;
+    pub const FRAMEDELAYMAX: usize = 5;
+}
+
+mod subscribe_nr {
+    pub const TRANSMIT: usize = 1;
+    pub const RECEIVE: usize = 2;
+    pub const SELECT: usize = 3;
+}
+
+mod allow_nr {
+    pub const TRANSMIT: usize = 1;
+    pub const RECEIVE: usize = 2;
+}
+
+pub struct NfcTag {}
+
+impl NfcTag {
+    pub fn enable_emulation() {
+        NfcTag::emulate(true);
+    }
+
+    pub fn disable_emulation() {
+        NfcTag::emulate(false);
+    }
+
+    pub fn emulate(enabled: bool) -> bool {
+        let result_code =
+            syscalls::command(DRIVER_NUMBER, command_nr::EMULATE, enabled as usize, 0);
+        if result_code.is_err() {
+            return false;
+        }
+
+        true
+    }
+
+    /// Subscribe to the tag being SELECTED callback.
+    pub fn selected() -> bool {
```

Comment:
You could consider not returning `bool` everywhere, but just passing on the error code with a `Result<(), ErrorType>`. The you'd have access to the `?` operator below, like
`syscalls::subscribe(...)?;
instead of `is_err`.

Reply:
Actually, libtock-rs has its own return type which should be used: `TockResult`

Reply:
Ah yeah, only saw your comment now. Please consider my suggestions below with `TockResult`.